### PR TITLE
Aggregated code

### PR DESCRIPTION
**Pull request by Fraseer**
Three different branches:
1. Application Structure/Testing Framework
2. Visitor can navigate through the website 
3. Visitor can view a collection of products (Only cypress tests have been worked on)

PT Links:
1. https://www.pivotaltracker.com/story/show/179716940 , https://www.pivotaltracker.com/story/show/179716943
2. https://www.pivotaltracker.com/story/show/179720015
3. https://www.pivotaltracker.com/story/show/179716947
------

1. Application structure
- Clean up unnecessary code
- Configured Cypress & Jest as testing framework 

2. Visitor can navigate through the website
- Tests for moving between the different tabs for the website.
- Adds the following components to give the website its basic layout:
     - Home Page
     - Menu Page
     - Contact Page
     - Header
     - Footer
-  Content added to components is just for running tests
- Netlify deployment

3. Visitor can view a collection of products
-   Creates tests file and test for when first item is added.

### CODE REVIEW

File: cypress/integration/userCanSeeTheMenuList.feature.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    cy.get("[data-cy=menu]").click()
+  });
+
+  it('is expected to display the main menu', () => {
+    cy.get("[data-cy=main-menu]").should("contain", "Main Menu")
+  });
+
+  it('is expected to show the first item on the menu', () => {
+    cy.get("[data-cy=item-1").within(() =>{
+      cy.get("[data-cy=id]").should("contain", 1)
```
Comment (tochman) [inline cypress/integration/userCanSeeTheMenuList.feature.js]: Why are you displaying the ID to the end-user?

File: cypress/integration/userCanSeeTheMenuList.feature.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      cy.get("[data-cy=dish]").should("contain","Kangaroo Steak")
+      cy.get("[data-cy=description]").should("contain", "Bouncy bouncy")
+      cy.get("[data-cy=price]").should("contain", "500 kr")
+      // a picture?
```
Comment (tochman) [inline cypress/integration/userCanSeeTheMenuList.feature.js]: Please remove comments. Add new features like adding an image to your PivotalTracker rather then making a comment in the code

File: src/components/HomePage.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+const HomePage = () => {
+  return(
+  <Container>
+    <Header size="huge" color="red" data-cy="resturant-name">To Gross For Comfort?</Header>
+  </Container>
+  );
+};
+
+export default HomePage;
```
Comment (tochman) [inline src/components/HomePage.jsx:14]: Be mindful of code formatting

File: cypress/integration/userCanSeeTheMenuList.feature.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  });
+
+  it('is expected to display the main menu', () => {
+    cy.get("[data-cy=main-menu]").should("contain", "Main Menu")
+  });
+
+  it('is expected to show the first item on the menu', () => {
+    cy.get("[data-cy=item-1").within(() =>{
+      cy.get("[data-cy=id]").should("contain", 1)
+      cy.get("[data-cy=dish]").should("contain","Kangaroo Steak")
```
Comment (tochman) [inline cypress/integration/userCanSeeTheMenuList.feature.js]: This test is not passing for you, right? We can't merge PR's with failing tests. Either remove the rest or implement the functionality

File: src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import Footer from "./components/Footer";
+import HomePage from "./components/HomePage";
+import Menu from './components/Menu'
+import About from './components/About'
+
+const App = () => {
+  return (
+    <BrowserRouter>
+      <Header />
+
```
Comment (tochman) [inline src/App.jsx]: No need for empty lines

**Review by tochman — COMMENTED** [review]
Not quite ready to merge. Please review my comments.

File: cypress/integration/userCanSeeTheMenuList.feature.js (hunk 1 above)
Comment (dorianbuck) [reply cypress/integration/userCanSeeTheMenuList.feature.js]: removed test for visible ID

File: cypress/integration/userCanSeeTheMenuList.feature.js (hunk 2 above)
Comment (dorianbuck) [reply cypress/integration/userCanSeeTheMenuList.feature.js]: Comment removed

File: src/components/HomePage.jsx (hunk 3 above)
Comment (dorianbuck) [reply src/components/HomePage.jsx:14]: Component formatted

File: cypress/integration/userCanSeeTheMenuList.feature.js (hunk 4 above)
Comment (dorianbuck) [reply cypress/integration/userCanSeeTheMenuList.feature.js]: Test file removed for now

File: src/App.jsx (hunk 5 above)
Comment (dorianbuck) [reply src/App.jsx]: Empty line removed

**Comment by dorianbuck** [comment]
We have implemented these changes and pushed them to this branch

**Pull request merged**